### PR TITLE
fix: resolve parameter validation bug in project-discovery tools

### PR DESCRIPTION
**Pull request by cameroncooke**
## Summary
Fixes parameter validation bug in three project-discovery tools where tools were receiving `{signal:{}, requestId:1}` instead of actual user parameters, and updates Reloaderoo documentation to reflect latest v1.1.2+ changes.

### Bug Fix Root Cause Analysis:
- **Root Cause**: Tools using `z.object({...})` schema pattern were not receiving parameters correctly from MCP clients
- **Specific Conditions**: Affected `list_schems_ws`, `show_build_set_proj`, and `show_build_set_ws` tools when called via Cursor AI or other MCP clients
- **Why Current Code Failed**: The `z.object()` wrapper in schema definitions prevented proper parameter parsing, causing tools to receive internal MCP protocol objects instead of user parameters

### Documentation Updates:
- Reloaderoo documentation was outdated and referenced older CLI interface from pre-v1.1.2

### Bug Fix Technical Approach:
- Changed schema definitions from `z.object({param: z.string().describe('...')})` to plain object `{param: z.string().describe(...)}`
- Added proper parameter casting in `list_schems_ws` for safe property access: `const paramsRecord = params as Record<string, unknown>;`
- Maintained all existing validation logic and error handling

### Documentation Updates:
- Added version reference (v1.1.2+) to both documentation files
- Updated command structure to match latest CLI interface
- Fixed info command options to use `-v, --verbose` format
- Preserved important debug logging capabilities for development workflows
- Removed outdated `--raw` and per-command `--log-level` options that don't exist in v1.1.2

### Bug Fix Reproduction & Validation:
- **Reproduction Steps**: Called `list_schems_ws` via Cursor AI with valid parameters, received "Required parameter 'workspacePath' is missing" error despite parameter being provided
- **Validation Method**: Used Reloaderoo CLI to systematically test affected tools before and after fix
- **Manual Testing**: Verified all three tools (`list_schems_ws`, `show_build_set_proj`, `show_build_set_ws`) work correctly with the fix
- **Edge Cases**: Tested with various parameter combinations and verified error handling remains intact

### Documentation Testing:
- Verified all command examples work with latest Reloaderoo v1.1.2
- Confirmed debug logging options are correctly documented
- Tested command structure matches actual CLI behavior

## Notes
- Bug affects only tools that used `z.object()` schema pattern - other tools with plain object schemas were unaffected
- No breaking changes to tool interfaces or functionality
- Documentation updates preserve development-critical debug logging capabilities
- Both changes are backward compatible and improve user experience

## Summary by CodeRabbit

* **New Features**
  * Introduced a comprehensive CLI tool for summarizing available tools and resources.
  * Added detailed manual testing guidelines and agent specifications for black-box QA validation.
  * Expanded documentation on ESLint type safety rules and strict release process enforcement.
  * Enhanced CLI documentation for clarity and explicit versioning.

* **Bug Fixes**
  * Improved type safety and error handling across many commands and utilities.
  * Standardized default value handling using nullish coalescing for more predictable behavior.
  * Strengthened validation and runtime checks for device and simulator data parsing.

* **Refactor**
  * Replaced generic parameter types with explicit, strongly-typed interfaces throughout the codebase.
  * Updated internal logic for parameter validation, error responses, and command construction for clarity and safety.

* **Tests**
  * Updated test assertions and mocks to align with stricter type safety and new parameter structures.
  * Enhanced test coverage for new and refactored features.

* **Chores**
  * Added and updated npm scripts for tool/resource discovery.
  * Improved CI workflow with automated type checks and stricter linting.

* **Documentation**
  * Added and revised multiple documents to guide testing, release, and code quality processes.
  * Updated CLI usage examples and troubleshooting instructions for end-users.

### CODE REVIEW

**Comment by cameroncooke** [comment]
Cursor review

**Comment by cameroncooke** [comment]
cursor review

File: docs/RELOADEROO.md [hunk 1]
```diff
@@ -139,15 +139,19 @@ When running under Claude Code, XcodeBuildMCP automatically detects the environm
...
 
 Commands:
-  proxy [options] -- <command>    🔄 Run as MCP proxy server (hot-reload mode)
-  inspect [subcommand]             🔍 Inspect and debug MCP servers (CLI mode)
-  info [options]                   📊 Display version and configuration information
-  help [command]                   ❓ Display help for command
+  proxy [options]  🔄 Run as MCP proxy server (default behavior)
+  inspect          🔍 Inspect and debug MCP servers
+  info [options]   📊 Display version and configuration information
+  help [command]   ❓ Display help for command
```
[the remark this answers is not in the document]
Comment (cameroncooke) [reply docs/RELOADEROO.md]: Nope you're being fussy

**Comment by cameroncooke** [comment]
cursor review

**Comment by cameroncooke** [comment]
cursor review
